myfitnesspal/client.py

```diff
@@ -734,3 +738,141 @@ def _get_water(self, date):
 
     def __unicode__(self):
         return u'MyFitnessPal Client for %s' % self.effective_username
+
+    def get_food_search_results(self, query):
+        search_url = parse.urljoin(self.BASE_URL_SECURE, self.SEARCH_PATH)
+        document = self._get_document_for_url(search_url)
+        authenticity_token = document.xpath(
+            "(//input[@name='authenticity_token']/@value)[1]"
+        )[0]
+        utf8_field = document.xpath(
+            "(//input[@name='utf8']/@value)[1]"
+        )[0]
+
+        result = self.session.post(
+            search_url,
+            data={
+                'utf8': utf8_field,
+                'authenticity_token': authenticity_token,
+                'search': query,
+                'date': dt.datetime.today().strftime('%Y-%m-%d'),
+                'meal': '0',
+            }
+        )
+
+        # result.content is bytes so we decode it ASSUMING utf8 (which may be a
```

Comment:
I think this is a perfectly reasonable assumption! If you wanted to make it at least less likely to raise an exception should that assumption prove incorrect, you could provide the second parameter to `decode` -- https://docs.python.org/3/library/stdtypes.html#bytes.decode